internal/ast/ast.go

```diff
@@ -207,22 +207,30 @@ type (
 		Op  string // Operator. // TODO: This should be a concrete type.
 		Arg *Ident // identifier of the argument being incremented
 	}
+
+	// SubscriptExpr node represents a subscript expression
+	SubscriptExpr struct {
+		Position
+		Object *Ident // identifier of the array object
+		Index  *Ident // identifier that contains the index of the subscript
```

Comment:
The index will not always be a identifier, we can have this code for example: `list[10]`, in this case the index would be a BasicLit, so we should change this type to be an Expr.


This code `[10,20,30][1]` is also a valid javascript code, so the Object type should also be an Expr.